rig/machine_control/scp_connection.py

```diff
@@ -79,65 +125,167 @@ def send_scp(self, buffer_size, x, y, p, cmd, arg1=0, arg2=0, arg3=0,
             The packet that was received in acknowledgement of the transmitted
             packet.
         """
-        self.sock.settimeout(self.default_timeout + timeout)
-
-        # Construct the packet that will be sent
-        packet = packets.SCPPacket(
-            reply_expected=True, tag=0xff, dest_port=0, dest_cpu=p,
-            src_port=7, src_cpu=31, dest_x=x, dest_y=y, src_x=0, src_y=0,
-            cmd_rc=cmd, seq=self._seq, arg1=arg1, arg2=arg2, arg3=arg3,
-            data=data
-        )
+        # This is implemented as a single burst packet sent using the bursty
+        # interface.  This significantly reduces code duplication.
+        # Construct a callable to retain the returned packet for us
+        class Callback(object):
```

Comment:
No, because the received packet has to be stored somewhere.

In Python 3 I can do:
```python
packet = None
def callback(received_bytes):
    nonlocal packet
    packet = ...
```

But not in Python 2.